packages/client/src/services/ApiClient.ts

```diff
@@ -0,0 +1,29 @@
+import axios, { AxiosStatic } from 'axios'
+
+export class ApiClient {
+  constructor(private _client: AxiosStatic = axios) {
+    axios.interceptors.response.use(
+      (response) => response,
+      (error) => {
+        const win: Window = window
+        if (error.response.status === 401) {
+          win.location = '/'
+        } else if (error.response.status === 404) {
+          win.location = '/error404'
+        } else {
+          win.location = '/error'
+        }
+      }
+    )
+  }
+
+  private get client() {
+    return this._client
+  }
+
+  get defaultService() {
+    this._client.defaults.baseURL = process.env.REACT_APP_SERVICE_URL
```

Comment:
Мне кажется лучше использовать client()
```suggestion
    this.client().defaults.baseURL = process.env.REACT_APP_SERVICE_URL
```

Reply:
Тут вроде нельзя так делать, мы переназначаем _client.defaults.baseURL у конструктора. client() нужен просто для просмотра _client. Не для изменения. Если я всё верно понимаю.

Reply:
Я исхожу из такой мысли: если у нас есть метод для получения свойства, то лучше использовать его, а не обращаться к свойству напрямую. В случае изменения имени свойства нам нужно будет менять его только в двух местах в коде, вместо нескольких. Как-то так)